datadog_checks_dev/tests/test_kind.py

```diff
@@ -0,0 +1,50 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import mock
+import pytest
+import tenacity
+
+from datadog_checks.dev import RetryError
+from datadog_checks.dev.ci import running_on_ci
+from datadog_checks.dev.kind import kind_run
+
+from .common import not_windows_ci
+
+
+class TestKindRun:
+    @pytest.mark.parametrize(
+        "attempts,expected_call_count",
+        [
+            (None, 1),
+            (0, 1),
+            (1, 1),
+            (3, 3),
+        ],
+    )
+    @not_windows_ci
+    def test_retry_on_failed_conditions(self, attempts, expected_call_count):
+        condition = mock.MagicMock()
+        condition.side_effect = RetryError("error")
+
+        expected_exception = tenacity.RetryError
+        if attempts is None:
+            if running_on_ci():
+                expected_call_count = 2
+            else:
+                expected_exception = RetryError
```

Comment:
Why is this needed?

Reply:
If no retry are set, the function raise the original PR. If a retry is specified, we do not raise the original exception but a `tenacity.RetryError`.

Your other comment is very relevant and it makes it harder to understand the test. I updated it with Exception. 

Also, I think it's not that great that we hide the actual exception with a `tenacity.RetryError`. I think we could modify this in a follow up PR. What do you think?

Reply:
Oh, what I meant to ask was why we need to do something differently depending on whether we are on CI.

Reply:
Because the retry is forced to 2 if the tests are running on the CI 

https://github.com/DataDog/integrations-core/blob/f07ee43878a06566bd6f63441e2298a9cd31a23f/datadog_checks_dev/datadog_checks/dev/env.py#L88-L89

Reply:
Ohhhh I get it now, I wish that was handled at a different level instead of having to rely on global state.